src/lib/eliom_common.server.ml

```diff
@@ -272,16 +272,18 @@ type 'a servicecookiestable = 'a servicecookiestablecontent SessionCookies.t
    - the group to which belongs the session
 *)
 
-(* non persistent cookies for in memory data *)
-type datacookiestablecontent =
-  (full_state_name *
-   float option ref        (* expiration date by timeout
-                              (server side) *) *
-   timeout ref             (* user timeout *) *
-   cookie_level sessgrp ref   (* session group *) *
-   string Ocsigen_cache.Dlist.node (* session group node *))
+module Data_cookie = struct
+  (* non persistent cookies for in-memory data *)
+  type t = {
+    full_state_name : full_state_name;
+    expiry : float option ref;
+    timeout : timeout ref;
+    session_group : cookie_level sessgrp ref;
```

Comment:
Could we use mutable fields instead?

Reply:
Unfortunately not. Not easily in any case. In `src/lib/server/eliommod_cookies.ml` for instance we pass on these references to other records:
```
                        {Eliom_common.dc_hvalue= hvalue       (* value *);
                         Eliom_common.dc_set_value= None;
                         Eliom_common.dc_timeout= timeout
                          (* user timeout ref *);
                         [...]
                         Eliom_common.dc_session_group= session_group;
                         Eliom_common.dc_session_group_node= session_group_node}
```

These references are then modified in other contexts :
```
src/lib/eliom_state.server.ml:    c.Eliom_common.dc_session_group := n;
src/lib/server/eliommod_datasess.ml:            c.Eliom_common.dc_session_group := fullsessgrp
```

I'm open to change these into mutable fields, but I couldn't do so without understanding the code much better than I currently do. Maybe after some more (more trivial) refactoring I will be able to.

Reply:
Ok